x-pack/plugin/ml-package-loader/src/main/java/org/elasticsearch/xpack/ml/packageloader/action/ModelLoaderUtils.java

```diff
@@ -114,32 +116,36 @@ static InputStream getInputStreamFromModelRepository(URI uri) throws IOException
         }
     }
 
-    static Tuple<List<String>, List<String>> loadVocabulary(URI uri) {
-        try {
-            InputStream vocabInputStream = getInputStreamFromModelRepository(uri);
-
-            if (uri.getPath().endsWith(".json")) {
-                XContentParser sourceParser = XContentType.JSON.xContent()
-                    .createParser(
-                        XContentParserConfiguration.EMPTY,
-                        Streams.limitStream(vocabInputStream, VOCABULARY_SIZE_LIMIT.getBytes())
-                    );
-                Map<String, List<Object>> vocabAndMerges = sourceParser.map(HashMap::new, XContentParser::list);
-
-                List<String> vocabulary = vocabAndMerges.containsKey(VOCABULARY)
-                    ? vocabAndMerges.get(VOCABULARY).stream().map(Object::toString).collect(Collectors.toList())
-                    : Collections.emptyList();
-                List<String> merges = vocabAndMerges.containsKey(MERGES)
-                    ? vocabAndMerges.get(MERGES).stream().map(Object::toString).collect(Collectors.toList())
-                    : Collections.emptyList();
-
-                return Tuple.tuple(vocabulary, merges);
+    static VocabularyParts loadVocabulary(URI uri) {
+        if (uri.getPath().endsWith(".json")) {
+            try {
+                InputStream vocabInputStream = getInputStreamFromModelRepository(uri);
```

Comment:
We weren't doing this prior to this PR but should we use a try-with-resources to wrap the inputstream and ensure it gets closed if an exception occurs?

Reply:
Good point, I've made that change